### PR TITLE
Move pay tabs to pages with routes, Add pay settings page in project layout

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on restructuring the Pay section in the dashboard, integrating API key functionalities, and improving error handling.

### Detailed summary
- Deleted `PayAnalytics.tsx` and `PayUI` components
- Added new components and pages for API key management
- Updated text in `ProjectGeneralSettingsPage.tsx`
- Improved error handling in API key fetching functions

> The following files were skipped due to too many changes: `apps/dashboard/src/app/(dashboard)/dashboard/connect/pay/page.tsx`, `apps/dashboard/src/components/pay/PayConfig.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
* **#4658** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4658?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/pay/[id]/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { TabPathLinks } from "@/components/ui/tabs";
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/pay/[id]/layout.tsx:1]: Before merging - verify that "Recipient address" input in settings tab is updating the given input properly ( do page reload and check ) 

For some reason, it's not working on locahost

Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/pay/[id]/layout.tsx:1]: It's a bug in api server 

If I fetch a single api key using /v1/keys/<id> - it doesn't include the payoutAddress in the pay service object
but If I get all the api keys using /v1/keys/ its included

Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/pay/[id]/layout.tsx:1]: added a workaround to always use  /v1/keys/ endpoint for now

**Review by jnsdls — APPROVED** [review]

**Pull request merged**